frontend/src/components/MapTooltip/MapTooltip.tsx

```diff
@@ -10,43 +14,89 @@ type MapTooltipProps = {
   coordinates: LngLat
 }
 
-let MapTooltip: React.StatelessComponent<MapTooltipProps> = ({ children, coordinates }) =>
-  <Popup
-    anchor='bottom'
-    className='MapTooltip'
-    coordinates={coordinates.toArray()}
-    offset={[0, -20]}
-  >
-    <table>
-      <tbody>
-        {children}
-      </tbody>
-    </table>
-  </Popup>
-
-type Props = {
+let MapTooltip: React.StatelessComponent<MapTooltipProps> = ({ children, coordinates }) => {
+  return (
+    <Popup
+      anchor='bottom'
+      className='MapTooltip'
+      coordinates={coordinates.toArray()}
+      offset={[0, -20]}
+    >
+      {children}
+    </Popup>
+  )
+}
+
+type ProviderProps = {
+  store: Store
+  point: GeoJSONEventData
+}
+
+type RepresentativePointProps = {
   point: GeoJSONEventData
 }
 
+function _ToggleNextProvider(store: Store, maxIndex: number) {
+  console.log('toggle next')
+  let index = store.get('providerIndex')
+  if (store.get('providerIndex') === maxIndex)
+    return
+  store.set('providerIndex')(index + 1)
+}
+
+function _TogglePrevProvider(store: Store) {
+  console.log('toggle prev')
+  let index = store.get('providerIndex')
+  if (index === 0)
+    return
+  store.set('providerIndex')(index - 1)
+}
+
 /**
  * TODO: expose info for all providers with the same address,
  * not only the first one.
  */
-export let ProviderPopup: React.StatelessComponent<Props> =
-  ({ point: { features, lngLat } }) =>
-    <MapTooltip coordinates={lngLat}>
-      <TableRow name='Address' value={features[0].properties.address} />
-      <TableRow name='Specialty' value={features[0].properties.specialty} />
-      <TableRow name='Lat' value={formatCoordinate(lngLat.lat)} />
-      <TableRow name='Lng' value={formatCoordinate(lngLat.lng)} />
-    </MapTooltip>
+export let ProviderPopup: React.StatelessComponent<ProviderProps> =
```

Comment:
Instead of passing in `store` as a prop, use `withStore`:

```diff
+ import { StoreProps } from '../store'

type ProviderProps = {
-  store: Store
  point: GeoJSONEventData
}

- export let ProviderPopup: React.StatelessComponent<ProviderProps> =
+ export let ProviderPopup = withStore('providerIndex')<StoreProps & ProviderProps>(({ store }) =>
```

Reply:
updated, let me know if it looks OK now.